### DISCUSSION TITLE
/people.districtID ?

### DISCUSSION THREAD

**Discussion by stevenromalewski** [openstates/issues#288] [Q&A]
We're just starting to use the v3 API. There's an option to identify legislators and their office contact info via the /people.geo API call.  But that requires a lat/lon, as far as we can tell.  Maybe we're missing something obvious, but is there an API call that will retrieve legislators and their office contact info via a district ID?

We used an earlier version of the API at our Census 2020 Hard to Count/Response Rate map https://www.censushardtocountmaps2020.us/ and it was terrific. It really helped census stakeholders visualize the implications of the 2020 census for state legislative districts (and connect these stakeholders with state elected officials).

**Comment by jamesturk** [comment]
Hi Steven,

Thanks for getting in touch.

When you say districtID, what are you referring to?

We have a way to get legislators by state (jurisdiction) — most just get a whole state’s worth and don’t iterate over districts (which is tricky for a number of reasons) but if you have a need we can consider it.

James

**Reply by stevenromalewski** [reply]
Thanks James. What we're hoping to do is be able to select a representative's contact info just based on the district ID. The use case is for our web app, we'll have a drop-down list of district IDs from which a user can select one, and then we'd send that selected district ID to the OpenStates API to then display contact info.

By "district ID" I mean the unique ID number for a state upper or lower district, which can be concatenated with state FIPS code, or on its own as long as we can limit the search via the API to a specific state. So, in New York, the state Assembly District 52 would either be 36052, or just 52 as long as we can limit the API search to all lower house districts in state 36.

Or, in Massachusetts for the 11th Hampden District it would be "11th Hampden" or something like "25_11th Hampden" or if we use the Census Bureau's FIPS code for lower house districts in Massachusetts it would be 25112.

The Census Bureau FIPS codes for state legislative districts might be good to add to the API.  Then anyone who is working with Census Bureau demographic data and/or boundaries for state lower/upper districts could easily integrate the OpenStates data with the Census Bureau info.

**Reply by stevenromalewski** [reply]
Btw, it looks like the OpenStates API used to have a district ID, because at our Census 2020 mapping site https://www.censushardtocountmaps2020.us/ we have a search option for state legislative districts (upper right corner of the map). When someone selects a district we display various census-related info for the district in the left-hand panel at that link. We also connect with an earlier version of the OpenStates API to display legislative contact info.  That used to work ok, but now the section of the left-hand panel just has a loading spinner; we never get the data back from the API, presumably because the district IDs that at one time we had downloaded via OpenStates are no longer working?  See screenshot below (or at this shared Dropbox link: https://www.dropbox.com/s/xfzohckmfrmqaj4/census2020_openstates_spinner.png?dl=0 ):
![census2020_openstates_spinner](https://user-images.githubusercontent.com/49545563/108554835-ac028400-72c2-11eb-83be-c22f52ba8b00.png)

**Reply by stevenromalewski** [reply]
We had hoped to use a workaround, but I don't think this will work as I describe below.

We thought we could use the open states legislator IDs (such as ocd-person/8f685697-5fb0-4966-8cd0-ad3059650f8d ) and assign those to the district IDs we have in our geographic data from the Census Bureau. But it looks like for states that have multi-member districts such as New Hampshire, that won't work because the OpenStates legislator IDs are not unique to a district.  (There are 398 entries in your bulk download data for New Hampshire's lower house, but only 160 actual lower house districts in NH.)

**Comment by jamesturk** [comment]
Adding FIPS would be interesting, and something to consider for sure.  

In the interim, it wouldn't be too hard to add a district= parameter to the people endpoint that would allow searching by district name/number.

If you have a chance before I get to it, can you open a feature request ticket?  I'll find some time soon to squeeze this in, I don't think it'd be a major lift and the use case makes sense.

**Comment by jamesturk** [comment]
I had some time, so squeezed in the district= parameter -- you can see it here: https://v3.openstates.org/docs#/people/people_search_people_get

**Reply by stevenromalewski** [reply]
Terrific, thanks, I'll take a look!
